src/main/scala/scripts/s3wrapper/S3RemoteResource.scala

```diff
@@ -0,0 +1,52 @@
+package scripts.s3wrapper
+
+import scala.annotation.tailrec
+
+sealed trait S3RemoteResource {
+  def key: String
+  def sizeInBytes: Long
+}
+
+case class S3RemoteFile(key: String, sizeInBytes: Long) extends S3RemoteResource
+
+case class S3RemoteDirectory(key: String, children: Set[S3RemoteResource] = Set.empty) extends S3RemoteResource {
```

Comment:
I wasn't able to find usage of S3RemoteDirectory or S3RemoteResource. Does that mean that we can leave only S3RemoteFile?

Reply:
There were some other code that was removed for this PR. so, you are right. I am removing it.